source/game/field/KCollisionTypes.hh

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <egg/math/Vector.hh>
+
+#define KCL_ATTRIBUTE_TYPE(x) ((x)&0x1f)
+#define KCL_TYPE_BIT(x) (1 << (x))
+#define KCL_ATTRIBUTE_TYPE_BIT(x) KCL_TYPE_BIT(KCL_ATTRIBUTE_TYPE(x))
+
+// KCL attribute types
+typedef enum {
+    COL_TYPE_ROAD = 0,
+    COL_TYPE_SLIPPERY_ROAD = 1,
+    COL_TYPE_WEAK_OFF_ROAD = 2,
+    COL_TYPE_OFF_ROAD = 3,
+    COL_TYPE_HEAVY_OFF_ROAD = 4,
+    COL_TYPE_SLIPPERY_ROAD_2 = 5,
+    COL_TYPE_BOOST_PAD = 6,
+    COL_TYPE_BOOST_RAMP = 7,
+    COL_TYPE_JUMP_PAD = 8,
+    COL_TYPE_ITEM_ROAD = 9,
+
+    COL_TYPE_SOLID_OOB = 0xa,
+    COL_TYPE_MOVING_WATER = 0xb,
+
+    COL_TYPE_WALL = 0xc,
+    COL_TYPE_INVISIBLE_WALL = 0xd,
+    COL_TYPE_ITEM_WALL = 0xe,
+    COL_TYPE_WALL_2 = 0xf,
+
+    COL_TYPE_FALL_BOUNDARY = 0x10,
+    COL_TYPE_CANNON_TRIGGER = 0x11,
+    COL_TYPE_FORCE_RECALCULATE_ROUTE = 0x12,
+    COL_TYPE_HALFPIPE_RAMP = 0x13,
+    COL_TYPE_PLAYER_WALL = 0x14,
+    COL_TYPE_MOVING_ROAD = 0x15,
+    COL_TYPE_STICKY_ROAD = 0x16,
+    COL_TYPE_ROAD2 = 0x17,
+    COL_TYPE_SOUND_TRIGGER = 0x18,
+    COL_TYPE_WEAK_WALL = 0x19,
+    COL_TYPE_EFFECT_TRIGGER = 0x1a,
+    COL_TYPE_ITEM_STATE_MODIFIER = 0x1b,
+
+    COL_TYPE_HALFPIPE_INVISIBLE_WALL = 0x1c,
+    COL_TYPE_ROTATING_ROAD = 0x1d,
+    COL_TYPE_SPECIAL_WALL = 0x1e,
+    COL_TYPE_INVISIBLE_WALL2 = 0x1f,
+
+    COL_TYPE_COUNT
+} KColType;
+
+#define KCL_TYPE_DIRECTIONAL \
+    (KCL_TYPE_BIT(COL_TYPE_FALL_BOUNDARY) | KCL_TYPE_BIT(COL_TYPE_SOUND_TRIGGER) | \
+            KCL_TYPE_BIT(COL_TYPE_FORCE_RECALCULATE_ROUTE) | \
+            KCL_TYPE_BIT(COL_TYPE_EFFECT_TRIGGER) | KCL_TYPE_BIT(COL_TYPE_CANNON_TRIGGER))
+
+#define KCL_TYPE_SOLID_SURFACE \
+    (0xffffffff & \
```

Comment:
Each of this `COL_TYPE_` values are within 4 bytes. We should be able to get rid of the `& 0xffffffff`